Libraries/src/Amazon.Lambda.RuntimeSupport/Bootstrap/LambdaBootstrap.cs

```diff
@@ -115,20 +116,29 @@ public async Task RunAsync(CancellationToken cancellationToken = default(Cancell
         {
             if(UserCodeInit.IsCallPreJit())
             {
-                InternalLogger.GetDefaultLogger().LogInformation("PreJit: CultureInfo");
+                this._logger.LogInformation("PreJit: CultureInfo");
                 UserCodeInit.LoadStringCultureInfo();
 
-                InternalLogger.GetDefaultLogger().LogInformation("PreJit: Amazon.Lambda.Core");
+                this._logger.LogInformation("PreJit: Amazon.Lambda.Core");
                 UserCodeInit.PreJitAssembly(typeof(Amazon.Lambda.Core.ILambdaContext).Assembly);
             }
 
+            // For local debugging purposes this environment variable can be set to run a Lambda executable assembly and process one event
+            // and then shut down cleanly. Useful for profiling or running local tests with the .NET Lambda Test Tool. This environment
+            // variable should never be set when function is deployed to Lambda.
+            var runOnce = string.Equals(Environment.GetEnvironmentVariable(Constants.ENVIRONMENT_VARIANLE_AWS_LAMBDA_DOTNET_DEBUG_RUN_ONCE), "true", StringComparison.OrdinalIgnoreCase);
+
             bool doStartInvokeLoop = _initializer == null || await InitializeAsync();
 
             while (doStartInvokeLoop && !cancellationToken.IsCancellationRequested)
             {
                 try
                 {
                     await InvokeOnceAsync(cancellationToken);
+                    if(runOnce)
+                    {
+                        return;
```

Comment:
We should do some logging that Lambda is configured for run once and process is exiting for diagnostics.

Reply:
Good idea, I have added the messaging.